test-data/unit/check-expressions.test

```diff
@@ -936,8 +936,8 @@ main:3: error: "A" not callable
 
 a, o = None, None # type: (A, object)
 a = f()         # E: "f" does not return a value
-o = a()         # E: Function does not return a value
-o = A().g(a)    # E: "g" of "A" does not return a value
+o = a()
```

Comment:
It would be okay for this to still generate an error, since `__call__` has an explicit `None` return, but since this is an edge case it's not that important. Might be worth adding an issues about this though.